lua/lint/linters/spectral.lua

```diff
@@ -0,0 +1,56 @@
+local severities = {
+  vim.diagnostic.severity.HINT,
+  vim.diagnostic.severity.INFO,
+  vim.diagnostic.severity.WARN,
+  vim.diagnostic.severity.ERROR,
+}
+
+local no_ruleset_found_msg = [[Spectral could not find a ruleset.
+Please define a .spectral file in the current working directory or override the linter args to provide the path to a ruleset.]]
+
+return {
+  cmd = 'spectral',
+  stdin = false,
+  append_fname = true,
+  args = { "lint", "-f", "json", },
+  stream = "both",
+  ignore_exitcode = true,
+  parser = function(output, _)
+    -- inform user if no ruleset has been found
+    if string.find(output, "No ruleset has been found") ~= nil then
+      vim.notify(no_ruleset_found_msg, vim.log.levels.WARN)
+      return {}
+    end
+
+    -- attempt to decode linter result
+    local success, result = pcall(vim.json.decode, output)
+    if not success then
+      vim.notify("Spectral exited with an error:\n" .. output, vim.log.levels.ERROR)
+      return {}
+    end
```

Comment:
```suggestion
    result = vim.json.decode(output)
```

You can just let it fail, there is default error handling: https://github.com/mfussenegger/nvim-lint/blob/03b1fc593638098a35de26d768d5f43b0fe57041/lua/lint/parser.lua#L143-L155